src/Umbraco.Web.UI.Client/src/common/directives/validation/valservermatch.directive.js

```diff
@@ -40,41 +52,63 @@ function valServerMatch(serverValidationManager) {
             
             var unsubscribe = [];
 
-            //subscribe to the server validation changes
-            function serverValidationManagerCallback(isValid, propertyErrors, allErrors) {
-                if (!isValid) {
-                    formCtrl.$setValidity('valServerMatch', false);                    
-                }
-                else {
-                    formCtrl.$setValidity('valServerMatch', true);                    
+            function bindCallback(validationKey, matchVal, matchType) {
+
+                if (Utilities.isString(matchVal)) {
+                    matchVal = [matchVal]; // normalize to an array since the value can also natively be an array
                 }
+
+                // match for each string in the array
+                matchVal.forEach(m => {
```

Comment:
I've experience that 'matchVal' is undefined, could you please see if it make sense to add a undefined check? :-) @Shazwazza